### ISSUE TITLE
support loading files from uri

### ISSUE THREAD

**Issue by mnahkies** [mnahkies/openapi-code-generator#43]
Implement support for loading definition files from URI.

Should be supported for both root/entry-point file, and also for referencing external schemas, eg: 
```
$ref: 'https://example.com/openapi.yaml#/components/responses/ClientError'
```

Care needs to be taken as to relative path handling. Eg: if you reference a external file using a URL, and it references a file relatively then this should be resolved relative to the URL.

See https://datatracker.ietf.org/doc/html/draft-pbryan-zyp-json-ref-03 for exact specifications.

https://github.com/mnahkies/openapi-code-generator/blob/ffc8f20487c11f339d08c2493ba68b23fea041f9/packages/openapi-code-generator/src/core/file-loader.ts#L46

**Comment by mnahkies** [comment]
Initial support released in [0.9.0](https://github.com/mnahkies/openapi-code-generator/releases/tag/v0.9.0)

It probably doesn't follow links to other schemas properly yet, but for standalone schemas will work fine.

**Comment by mnahkies** [comment]
Additionally added support for authenticated remote URI's in #230
